File: tests/java-rosettacode/Hanoi.java
public class Hanoi {
    public static void move(int n, String from, String to, String via) {
        if (n == 1) {
            System.out.printf("Move disk 1 from %s to %s%n", from, to);
            return;
        }
        move(n - 1, from, via, to);
        System.out.printf("Move disk %d from %s to %s%n", n, from, to);
        move(n - 1, via, to, from);
    }

    public static void main(String[] args) {
        int disks = 4;
        System.out.println("Towers of Hanoi with " + disks + " disks:");
        move(disks, "A", "C", "B");
    }
}

File: tests/rosettacode/hanoi.go
package main

import "fmt"

func hanoi(n int, from, to, via string) {
	if n == 1 {
		fmt.Printf("Move disk 1 from %s to %s\n", from, to)
		return
	}
	hanoi(n-1, from, via, to)
	fmt.Printf("Move disk %d from %s to %s\n", n, from, to)
	hanoi(n-1, via, to, from)
}

func main() {
	hanoi(4, "A", "C", "B")
}
